app/src/main/java/org/groundplatform/android/repository/SurveyRepository.kt

```diff
@@ -75,14 +77,18 @@ constructor(
     _selectedSurveyId.update { surveyId }
 
     // Wait for survey to be updated. Else throw an error after timeout.
-    withTimeout(ACTIVATE_SURVEY_TIMEOUT_MILLS) {
-      activeSurveyFlow.first { survey ->
-        if (surveyId.isBlank()) {
-          survey == null
-        } else {
-          survey?.id == surveyId
+    try {
+      withTimeout(ACTIVATE_SURVEY_TIMEOUT_MILLS) {
+        activeSurveyFlow.first { survey ->
+          if (surveyId.isBlank()) {
+            survey == null
+          } else {
+            survey?.id == surveyId
+          }
         }
       }
+    } catch (e: TimeoutCancellationException) {
+      Timber.e("Waiting for survey to get updated")
```

Comment:
Since the coroutine has already cancelled, let's fix the error message to `Failed to get survey due to timeout`. Also, add the exception to the timber to get the stacktrace as well